game-core/src/main/java/games/strategy/engine/chat/Chat.java

```diff
@@ -95,12 +92,13 @@ public void participantAdded(final ChatParticipant chatParticipant) {
     }
     chatters.put(chatParticipant, "");
     updateConnections();
-    for (final IChatListener listener : listeners) {
-      listener.addStatusMessage(chatParticipant.getPlayerName() + " has joined");
-      if (chatSoundProfile == ChatSoundProfile.GAME_CHATROOM) {
-        ClipPlayer.play(SoundPath.CLIP_CHAT_JOIN_GAME);
-      }
-    }
+    chatMessageListeners.forEach(
+        listener -> {
+          listener.addStatusMessage(chatParticipant.getPlayerName() + " has joined");
+          if (chatSoundProfile == ChatSoundProfile.GAME_CHATROOM) {
+            ClipPlayer.play(SoundPath.CLIP_CHAT_JOIN_GAME);
+          }
```

Comment:
I had a look at the code where chatSoundProfile is being set, and it looks like it depends on the usage type this chat instance is supposed to be used for.
Any ideas how this information might relate to the kind of listeners registered?
If it turns out that specific listeners are only registered when the sound profile is for the chatroom, then this ClipPlayer invokation could easily be moved into the appropriate listeners.
Otherwise we might want to consider passing the chatSoundProfile so the listener can decide what to do.

Reply:
There even is an `addMessageWithSound` method, this is so weird.

Reply:
IMO it should really be the listeners decision to play sound or not, obviously the class should pass the "kind" of message to the listeners so they can decide based on that.

Reply:
Sounds about right @RoiEXLab . If we push the play sound functionality to listeners and in more appropriate places, I would guess eventually the `ChatSoundProfile` may no longer even be needed .